contracts/ethregistrar/StablePriceOracle.sol

```diff
@@ -4,86 +4,138 @@ import "./PriceOracle.sol";
 import "./SafeMath.sol";
 import "./StringUtils.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
 
 interface AggregatorInterface {
-  function latestAnswer() external view returns (int256);
+    function latestAnswer() external view returns (int256);
 }
 
-
 // StablePriceOracle sets a price in USD, based on an oracle.
-contract StablePriceOracle is Ownable, PriceOracle {
+contract StablePriceOracle is PriceOracle {
     using SafeMath for *;
     using StringUtils for *;
 
-    // Rent in base price units by length. Element 0 is for 1-length names, and so on.
-    uint[] public rentPrices;
+    // Rent in base price units by length
+    uint256 public immutable price1Letter;
+    uint256 public immutable price2Letter;
+    uint256 public immutable price3Letter;
+    uint256 public immutable price4Letter;
+    uint256 public immutable price5Letter;
 
     // Oracle address
     AggregatorInterface public immutable usdOracle;
 
-    event OracleChanged(address oracle);
-
-    event RentPriceChanged(uint[] prices);
-
-    bytes4 constant private INTERFACE_META_ID = bytes4(keccak256("supportsInterface(bytes4)"));
-    bytes4 constant private ORACLE_ID = bytes4(keccak256("price(string,uint256,uint256)") ^ keccak256("premium(string,uint256,uint256)"));
+    event RentPriceChanged(uint256[] prices);
 
-    constructor(AggregatorInterface _usdOracle, uint[] memory _rentPrices) public {
+    constructor(AggregatorInterface _usdOracle, uint256[] memory _rentPrices) {
         usdOracle = _usdOracle;
-        setPrices(_rentPrices);
+        price1Letter = _rentPrices[0];
+        price2Letter = _rentPrices[1];
+        price3Letter = _rentPrices[2];
+        price4Letter = _rentPrices[3];
+        price5Letter = _rentPrices[4];
     }
 
-    function price(string calldata name, uint expires, uint duration) external view override returns(uint) {
-        uint len = name.strlen();
-        if(len > rentPrices.length) {
-            len = rentPrices.length;
+    function price(
+        string calldata name,
+        uint256 expires,
+        uint256 duration
+    ) external view override returns (uint256, uint256) {
+        uint256 len = name.strlen();
+        if (len > 5) {
+            len = 5;
         }
         require(len > 0);
-        
-        uint basePrice = rentPrices[len - 1].mul(duration);
-        basePrice = basePrice.add(_premium(name, expires, duration));
 
-        return attoUSDToWei(basePrice);
+        uint256 basePrice;
+
+        if (len == 1) {
+            basePrice = price1Letter * duration;
+        } else if (len == 2) {
+            basePrice = price2Letter * duration;
+        } else if (len == 3) {
+            basePrice = price3Letter * duration;
+        } else if (len == 4) {
+            basePrice = price4Letter * duration;
+        } else if (len == 5) {
```

Comment:
Um, what if the length is >5?

A unit test should definitely have caught this!

Reply:
```
if (len > 5) {
    len = 5;
}
```

We normalise the value at the top (technically you did this)

Reply:
:facepalm: